migen/build/xilinx/vivado.py

```diff
@@ -31,7 +31,7 @@ def _format_xdc(signame, resname, *constraints):
         fmt_r += "." + resname[2]
     r = " ## {}\n".format(fmt_r)
     for c in fmt_c:
-        r += c + " [get_ports " + signame + "]\n"
+        r += c + " [get_ports {" + signame + "}]\n"
```

Comment:
What is this for and does it belong in this PR?

Reply:
Signal names can contain characters that have special meaning in Tcl (.xdc files are Tcl scripts), e.g. `bus_name[1]`. Surrounding names with `{}` prevents Tcl interpreter from processing them as a code. Apparently indexed signal names work under Vivado even without `{}`, but they don't under Yosys/Symbiflow.

For details about `{}` see:
Vivado Design Suite Tcl Command Reference Guide (ug835), Ch 1, General Tcl Syntax Guidelines, Using Special Characters
https://www.xilinx.com/support/documentation/sw_manuals/xilinx2019_2/ug835-vivado-tcl-commands.pdf (p. 10)

I've moved this change to separate commit.